overlord/snapstate/snapstate.go

```diff
@@ -1396,12 +1482,17 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []mi
 				waitPrereq(ts, update.SnapBase())
 			}
 		}
-		// keep track of kernel/gadget udpates
+		// keep track of kernel/gadget updates
 		switch update.Type() {
 		case snap.TypeKernel:
 			kernelTs = ts
 		case snap.TypeGadget:
 			gadgetTs = ts
+		case snap.TypeBase, snap.TypeOS:
+			if update.InstanceName() == deviceCtx.Model().Base() {
```

Comment:
does this work for core? isn't base "" then?

Reply:
There's a bit of complication with core, so I've added a comment below. When core is part of the update, we implicitly add its tasks as dependency for all tasks from other snaps. In case of a kernel (using core base) and core update, we end up with kernel tasks waiting for all core tasks. Thus introducing additional wait in core's auto-connect on link-snap of the kernel creates a wait loop. I think that technically it would be possible to tweak the wait lists such that we can reorder the tasks to achieve a single reboot setup, but I left it out for now.

Reply:
As discussed with @pedronis we will limit the feature to UC18+ devices.